File: Assets/Code/Scripts/DoorScripts/HoleDoorScript.js
#pragma strict

class HoleDoor extends Door {
  
  function HoleDoor( d : GameObject, i : Transform, o : Transform ) {
    super( d, i, o );
  }
  
  function AnimateIn() {
    Level.currentLevel.MovePlayer( outBlocking.position );
    super();
  }
  
  function AnimateOut() {
    Level.currentLevel.MovePlayer( inBlocking.position );
    super();
  }
}

public var destDoor : GameObject;
public var inBlocking : Transform;
public var outBlocking : Transform;

public var door : HoleDoor;

function Awake() {
  door = new HoleDoor( destDoor, inBlocking, outBlocking );
}

function OnMouseOver() {
  if( Input.GetMouseButtonDown( 0 ) ) StartCoroutine( door.Enter() );
}

File: Assets/Code/Scripts/DoorScripts/HoleDoorScript.cs
using UnityEngine;
using System.Collections;

public class HoleDoorScript : MonoBehaviour {

  public class HoleDoor : Door {
  
    public HoleDoor( GameObject d, Transform i, Transform o ) : base( d, i, o ) {}
  
    public override void AnimateIn() {
      Level.currentLevel.MovePlayer( outBlocking.position );
      base.AnimateIn();
    }
  
    public override void AnimateOut() {
      Level.currentLevel.MovePlayer( inBlocking.position );
      base.AnimateOut();
    }
  }

  public GameObject destDoor;
  public Transform inBlocking;
  public Transform outBlocking;

  public HoleDoor door;

  public void Awake() {
    door = new HoleDoor( destDoor, inBlocking, outBlocking );
  }

  public void OnMouseOver() {
    if( Input.GetMouseButtonDown(0) ) StartCoroutine( door.Enter() );
  }
}